File: army/Attack/DefaultAttack.cpp
#include "DefaultAttack.h"

DefaultAttack::DefaultAttack() {}

DefaultAttack::~DefaultAttack() {}

void DefaultAttack::attack(Unit* attacker, Unit* enemy) {
    enemy->takePhysicalDamage(attacker->getDamage());
    enemy->counterAttack(enemy, attacker); 
} 

void DefaultAttack::counterAttack(Unit* attacker, Unit* enemy) {
    enemy->takePhysicalDamage(attacker->getDamage() / 2);
}

File: army/JavaArmy/src/com/maksklaban/armymax/attack/DefaultAttack.java
package com.maksklaban.armymax.attack;

import com.maksklaban.armymax.exceptions.*;
import com.maksklaban.armymax.state.*;
import com.maksklaban.armymax.units.*;

public class DefaultAttack extends NormalAttack {
    public DefaultAttack() {}

    public void attack(Unit attacker, Unit enemy) throws ArmyException {
        enemy.takePhysicalDamage(attacker.getDamage());
        enemy.counterAttack(enemy, attacker);
    }

    public void counterAttack(Unit attacker, Unit enemy) throws ArmyException {
        enemy.takePhysicalDamage(attacker.getDamage() / 2);
    }
}
